pkgs/tools/admin/awscli2/default.nix

```diff
@@ -62,6 +69,17 @@ with py.pkgs; buildPythonApplication rec {
     wcwidth
   ];
 
+  checkPhase = ''
+    export PATH=$PATH:$out/bin
+
+    # Remove/Skip the following tests due to errors
+    rm tests/unit/customizations/history/test_history.py
+    rm tests/functional/kinesis/test_remove_operations.py
+    rm tests/functional/lex/test_remove_operations.py
```

Comment:
What errors?

Reply:
```
======================================================================
ERROR: test_attach_history_handler (tests.unit.customizations.history.test_history.TestAttachHistoryHandler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/nix/store/5598mbzzi0k3ahb2n96hd42w1l1qpx1p-python3.8-mock-4.0.3/lib/python3.8/site-packages/mock/mock.py", line 1346, in patched
    return func(*newargs, **newkeywargs)
  File "/tmp/nix-build-awscli2-2.2.1.drv-1/source/tests/unit/customizations/history/test_history.py", line 49, in test_attach_history_handler
    attach_history_handler(session=mock_session, parsed_args=parsed_args)
  File "/tmp/nix-build-awscli2-2.2.1.drv-1/source/awscli/customizations/history/__init__.py", line 53, in attach_history_handler
    os.makedirs(os.path.dirname(history_filename))
  File "/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/lib/python3.8/os.py", line 213, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/lib/python3.8/os.py", line 213, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/lib/python3.8/os.py", line 213, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/lib/python3.8/os.py", line 223, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
-------------------- >> begin captured logging << --------------------
awscli.customizations.history: DEBUG: Enabling CLI history
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: test_subscribe_to_shard_removed (tests.functional.kinesis.test_remove_operations.TestKinesisRemoveOperations)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-awscli2-2.2.1.drv-1/source/tests/functional/kinesis/test_remove_operations.py", line 21, in test_subscribe_to_shard_removed
    self.assertIn(error_msg, result.stderr)
AssertionError: 'argument operation: Invalid choice, valid choices are:' not found in '  File "/nix/store/xf9gfgz5dbxzp5nj798bhs0156bglxa2-awscli2-2.2.1/bin/aws", line 2\n    export PATH=\'/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/bin:/nix/store/xf9gfgz5dbxzp5nj798bhs0156bglxa2-awscli2-2.2.1/bin:/nix/store/psvf2gqa960gdqvjdpc5a5wfsrm1vr69-python3.8-jmespath-0.10.0/bin:/nix/store/1zzcz9m7i9b16s5p1ajqafs0skisq463-python3.8-docutils-0.16/bin:/nix/store/ck3b1dgq72mnm92dcpcsdnj9c9s2pja0-python3.8-distro-1.5.0/bin:/nix/store/bji5fz1z3pnwqs0lrll0rcz6sm2xrq84-groff-1.22.4/bin:/nix/store/3zgfyx7k8zcm2fmivmq8sfx5fj3x124z-less-563/bin:/nix/store/4znwdazvik3r0gpp7fa0w88sqx9sj08a-python3.8-rsa-4.7.2/bin\'${PATH:+\':\'}$PATH\n           ^\nSyntaxError: invalid syntax\n'
-------------------- >> begin captured logging << --------------------
awscli.tests.integration: DEBUG: Running command: python /nix/store/xf9gfgz5dbxzp5nj798bhs0156bglxa2-awscli2-2.2.1/bin/aws kinesis subscribe-to-shard help
awscli.tests.integration: DEBUG: rc: 1
awscli.tests.integration: DEBUG: stdout:
awscli.tests.integration: DEBUG: stderr:   File "/nix/store/xf9gfgz5dbxzp5nj798bhs0156bglxa2-awscli2-2.2.1/bin/aws", line 2
    export PATH='/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/bin:/nix/store/xf9gfgz5dbxzp5nj798bhs0156bglxa2-awscli2-2.2.1/bin:/nix/store/psvf2gqa960gdqvjdpc5a5wfsrm1vr69-python3.8-jmespath-0.10.0/bin:/nix/store/1zzcz9m7i9b16s5p1ajqafs0skisq463-python3.8-docutils-0.16/bin:/nix/store/ck3b1dgq72mnm92dcpcsdnj9c9s2pja0-python3.8-distro-1.5.0/bin:/nix/store/bji5fz1z3pnwqs0lrll0rcz6sm2xrq84-groff-1.22.4/bin:/nix/store/3zgfyx7k8zcm2fmivmq8sfx5fj3x124z-less-563/bin:/nix/store/4znwdazvik3r0gpp7fa0w88sqx9sj08a-python3.8-rsa-4.7.2/bin'${PATH:+':'}$PATH
           ^
SyntaxError: invalid syntax

--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: test_start_conversation_removed (tests.functional.lex.test_remove_operations.TestLexV2RuntimeRemoveOperations)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-awscli2-2.2.1.drv-1/source/tests/functional/lex/test_remove_operations.py", line 21, in test_start_conversation_removed
    self.assertIn(error_msg, result.stderr)
AssertionError: 'argument operation: Invalid choice, valid choices are:' not found in '  File "/nix/store/xf9gfgz5dbxzp5nj798bhs0156bglxa2-awscli2-2.2.1/bin/aws", line 2\n    export PATH=\'/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/bin:/nix/store/xf9gfgz5dbxzp5nj798bhs0156bglxa2-awscli2-2.2.1/bin:/nix/store/psvf2gqa960gdqvjdpc5a5wfsrm1vr69-python3.8-jmespath-0.10.0/bin:/nix/store/1zzcz9m7i9b16s5p1ajqafs0skisq463-python3.8-docutils-0.16/bin:/nix/store/ck3b1dgq72mnm92dcpcsdnj9c9s2pja0-python3.8-distro-1.5.0/bin:/nix/store/bji5fz1z3pnwqs0lrll0rcz6sm2xrq84-groff-1.22.4/bin:/nix/store/3zgfyx7k8zcm2fmivmq8sfx5fj3x124z-less-563/bin:/nix/store/4znwdazvik3r0gpp7fa0w88sqx9sj08a-python3.8-rsa-4.7.2/bin\'${PATH:+\':\'}$PATH\n           ^\nSyntaxError: invalid syntax\n'
-------------------- >> begin captured logging << --------------------
awscli.tests.integration: DEBUG: Running command: python /nix/store/xf9gfgz5dbxzp5nj798bhs0156bglxa2-awscli2-2.2.1/bin/aws lexv2-runtime start-conversation help
awscli.tests.integration: DEBUG: rc: 1
awscli.tests.integration: DEBUG: stdout:
awscli.tests.integration: DEBUG: stderr:   File "/nix/store/xf9gfgz5dbxzp5nj798bhs0156bglxa2-awscli2-2.2.1/bin/aws", line 2
    export PATH='/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/bin:/nix/store/xf9gfgz5dbxzp5nj798bhs0156bglxa2-awscli2-2.2.1/bin:/nix/store/psvf2gqa960gdqvjdpc5a5wfsrm1vr69-python3.8-jmespath-0.10.0/bin:/nix/store/1zzcz9m7i9b16s5p1ajqafs0skisq463-python3.8-docutils-0.16/bin:/nix/store/ck3b1dgq72mnm92dcpcsdnj9c9s2pja0-python3.8-distro-1.5.0/bin:/nix/store/bji5fz1z3pnwqs0lrll0rcz6sm2xrq84-groff-1.22.4/bin:/nix/store/3zgfyx7k8zcm2fmivmq8sfx5fj3x124z-less-563/bin:/nix/store/4znwdazvik3r0gpp7fa0w88sqx9sj08a-python3.8-rsa-4.7.2/bin'${PATH:+':'}$PATH
           ^
SyntaxError: invalid syntax

--------------------- >> end captured logging << ---------------------
```

Reply:
It's running 27618 more tests than it was previously, but will try to resolve these 3 test failures in this PR.